### PR TITLE
[JUJU-1761] Developed the bash-based change-user-password test using the expect tool

### PR DESCRIPTION
**Pull request by anvial**
This PR adds the change-user-password test, which is an interactive command. This test uses expect tool and generates expect scripts. 

The `expect_that` function is introduced, its implementation inspired by the `wait_for` function. This function can be part of `includes` tools.

PS:
please, use the apt version of the `expect` tool to avoid access problems.
```
sudo apt install expect
```

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made

## QA steps

```sh
cd tests
./main.sh -v -p lxd user run_user_change_password
```

### CODE REVIEW

**Review by hmlanigan — CHANGES_REQUESTED** [review]
Check for dependencies needs to be updated to include `expect`: https://github.com/juju/juju/blob/cde164f1124781f102e3d71bba88547043a1812a/tests/main.sh#L233

Running QA step resulted in:
couldn't read file "/home/heather/work-test/src/github.com/juju/juju/tests/tmp.q7t/juju-change-user-password.exp": permission denied

File: tests/suites/user/login_password.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# expect_command the ability to work with interactive commands with expect tool.
+# The output is "OK", if tests passes successfully.
+# The expect_script argument is a expect script body.
+# The default timeout is 10 seconds.
+#
+# ```
+# wait_for <command> <expect_script> [<timeout>]
```
[the remark this answers is not in the document]
Comment (anvial) [reply tests/suites/user/login_password.sh]: fixed

Also, there is a suggestion to rename the command from `expect_command` to `expect_that`. That will make this command more understandable to developers and closer to `wait_for`.

**Comment by anvial** [comment]
@hmlanigan ,

Looks like the problem is in the snap version of the `expect` tool. 

so 

```
sudo snap remove expect
sudo apt install expect
```

should help.

**Review by hmlanigan — APPROVED** [review]
A small nit based on issues encountered with QA. ow LGTM

**Comment by anvial** [comment]
Discussed with Joe the question of additional messaging about `apt` version of the `expect` tool. Looks like we shouldn't worry about specifically detecting apt. The `expect` snap is used only like a "pipe tool", so it's no like someone is likely to satisfy a check with the snap-installed version accidentally.

**Pull request merged**